### PR TITLE
389, 383, filtering menu

### PR DESCRIPTION
**Pull request by rkzel**
I didn't remove sorting, I made it kind of work..

### CODE REVIEW

**Comment by Quazia** [comment]
How hard would it be to make the sort/search case-insensitive?

File: apps/projects/app/components/Shared/FilterBar/FilterBar.js [hunk 1]
```diff
@@ -156,7 +179,12 @@ class FilterBar extends React.Component {
           </FilterDropDown>
 
           <FilterDropDown caption="Labels" enabled={ Object.keys(filtersData.labels).length > 0}>
-            { Object.keys(filtersData.labels).map(
+            { Object.keys(filtersData.labels).sort(
```
Comment (ottodevs) [inline apps/projects/app/components/Shared/FilterBar/FilterBar.js]: This should be calculated outside the return and captured into a variable instead

File: apps/projects/app/components/Shared/FilterBar/FilterBar.js [hunk 2]
```diff
@@ -175,7 +203,12 @@ class FilterBar extends React.Component {
           </FilterDropDown>
 
           <FilterDropDown caption="Milestones" enabled={ Object.keys(filtersData.milestones).length > 0}>
-            { Object.keys(filtersData.milestones).map(
+            { Object.keys(filtersData.milestones).sort(
```
Comment (ottodevs) [inline apps/projects/app/components/Shared/FilterBar/FilterBar.js]: This should be calculated outside the return and captured into a variable instead

File: apps/projects/app/components/Shared/FilterBar/FilterBar.js [hunk 3]
```diff
@@ -192,29 +225,27 @@ class FilterBar extends React.Component {
...
-          </FilterMenuItem>
-          <FilterMenuItem key={'2'} onClick={this.noop} style={{ display: 'flex', alignItems: 'flex-start' }}>
-            <div><CheckButton onChange={this.noop} checked={false} /></div>
-            <ActionLabel>...deadline</ActionLabel>
-          </FilterMenuItem>
-          <FilterMenuItem key={'3'} onClick={this.noop} style={{ display: 'flex', alignItems: 'flex-start' }}>
-            <div><CheckButton onChange={this.noop} checked={false} /></div>
-            <ActionLabel>status</ActionLabel>
-          </FilterMenuItem>
+          { this.state.sortBy.map(s =>
```
Comment (ottodevs) [inline apps/projects/app/components/Shared/FilterBar/FilterBar.js]: This should be calculated outside the return and captured into a variable instead

File: apps/projects/app/components/Content/Issues.js [hunk 4]
```diff
@@ -75,6 +76,11 @@ class Issues extends React.PureComponent {
     this.setState(prevState => ({ filters, reload: !prevState.reload }))
   }
 
+  handleSorting = sortBy => {
+    // TODO: why is reload necessary?
+    this.setState(prevState => ({ sortBy, reload: !prevState.reload }))
```
[the remark this answers is not in the document]
Comment (rkzel) [reply apps/projects/app/components/Content/Issues.js]: Left for refactoring - for now it's doing something, i.e., causing re-rendering

File: apps/projects/app/components/Content/Issues.js [hunk 5]
```diff
@@ -198,6 +205,7 @@ class Issues extends React.PureComponent {
         issues={[]}
         issuesFiltered={[]}
         handleFiltering={this.handleFiltering}
+        handleSorting={this.handleSorting}
```
[the remark this answers is not in the document]
Comment (rkzel) [reply apps/projects/app/components/Content/Issues.js]: Fixed in the latest commit.

File: apps/projects/app/components/Shared/FilterBar/FilterBar.js [hunk 6]
```diff
@@ -137,7 +158,9 @@ class FilterBar extends React.Component {
         
         <Overflow>
           <FilterDropDown caption="Projects" enabled={ Object.keys(filtersData.projects).length > 0}>
-            { Object.keys(filtersData.projects).map(
+            { Object.keys(filtersData.projects).sort(
```
[the remark this answers is not in the document]
Comment (rkzel) [reply apps/projects/app/components/Shared/FilterBar/FilterBar.js]: Discussed, left for refactoring in the nearest future.

File: apps/projects/app/components/Shared/FilterBar/FilterBar.js (hunk 1 above)
Comment (rkzel) [reply apps/projects/app/components/Shared/FilterBar/FilterBar.js]: Discussed, left for refactoring in the nearest future.

File: apps/projects/app/components/Shared/FilterBar/FilterBar.js (hunk 3 above)
Comment (rkzel) [reply apps/projects/app/components/Shared/FilterBar/FilterBar.js]: Discussed, left for refactoring in the nearest future.

File: apps/projects/app/components/Shared/FilterBar/FilterBar.js (hunk 2 above)
Comment (rkzel) [reply apps/projects/app/components/Shared/FilterBar/FilterBar.js]: Discussed, left for refactoring in the nearest future.

File: apps/projects/app/components/Content/Issues.js [hunk 7]
```diff
@@ -252,10 +270,11 @@ class Issues extends React.PureComponent {
                   issues={issues}
                   issuesFiltered={issuesFiltered}
                   handleFiltering={this.handleFiltering}
+                  handleSorting={this.handleSorting}
                   activeIndex={this.props.activeIndex}
                 />
                 <IssuesScrollView>
-                  {shapeIssues(issuesFiltered).map(issue => (
+                  {shapeIssues(issuesFiltered).sort(currentSorter).map(issue => (
```
[the remark this answers is not in the document]
Comment (rkzel) [reply apps/projects/app/components/Content/Issues.js]: Discussed, left for refactoring in the nearest future.

**Pull request merged**